Remotion/Validation/UnitTests/RoleCollectors/AddingPropertyValidationRuleCollector1Test.cs

```diff
@@ -105,51 +123,151 @@ public void Create_PropertyDeclaredInBaseClass ()
     public void CreateValidationRule_IgnoresValidationMessagesForValidatorsWithoutADeferredInitializationValidationMessage ()
     {
+      var validationMessage = new DelegateBasedValidationMessage(() => "Test");
+
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(e => e.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns(validationMessage);
+
+      var result = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object);
+
+      var propertyValidators = ((IPropertyValidationRule)result).Validators;
+
+      Assert.That(propertyValidators, Is.Empty);
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_WithValidationMessageFactoryReturnsNull_ThrowsInvalidOperationException ()
     {
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(_ => _.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns((ValidationMessage)null);
+
+      Func<PropertyValidationRuleInitializationParameters, IPropertyValidator> validatorFactory = param => new NotNullValidator(param.ValidationMessage);
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+
+      Assert.That(
+          () => _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object),
+          Throws.InvalidOperationException
+              .With.Message.EqualTo("The IValidationMessageFactory did not return a result for NotNullValidator applied to property 'UserName' on "
+                                    + "type 'Remotion.Validation.UnitTests.TestDomain.Customer'."));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void CreateValidationRule_WithConditionNotNull_InitializesConditionForCreatedPropertyValidationRule ()
     {
+      var validationMessage = new DelegateBasedValidationMessage(() => "Test");
+
+      var validationMessageMock = new Mock<IValidationMessageFactory>();
+      validationMessageMock
+          .Setup(e => e.CreateValidationMessageForPropertyValidator(It.IsAny<NotNullValidator>(), _property))
+          .Returns(validationMessage);
+
+      Func<PropertyValidationRuleInitializationParameters, IPropertyValidator> validatorFactory = param => new NotNullValidator(param.ValidationMessage);
+      Func<Customer, bool> predicate = _ => true;
+
+      _addingPropertyValidationRuleCollector.RegisterValidator(validatorFactory);
+      _addingPropertyValidationRuleCollector.SetCondition(predicate);
+
+      var result = _addingPropertyValidationRuleCollector.CreateValidationRule(validationMessageMock.Object);
+
+      Assert.That(((PropertyValidationRule<Customer, string>)result).Condition, Is.SameAs(predicate));
     }
 
     [Test]
-    [Ignore("TODO RM-5906")]
     public void SetCondition ()
```

Comment:
Maybe SetCondition_SetsConditionToBeUsedByCreate? Vielleicht reicht der derzeitige Name aber eh auch.

Reply:
Es wird eigentlich nur die Condition Property in der Klasse gesetzt, daher glaub ich passt einfach nur SetCondition (HappyPath)